src/ILCompiler.CppCodeGen/src/CppCodeGen/CppWriter.cs

```diff
@@ -1757,7 +1760,7 @@ private void OutputTypeNode(IEETypeNode typeNode, NodeFactory factory, CppGenera
                 if (nodeType.IsDelegate)
                 {
                     typeDefinitions.AppendLine();
-                    typeDefinitions.Append(GetCodeForDelegate(nodeType));
+                    typeDefinitions.Append(GetCodeForDelegate(nodeType, generateTypeDef:(virtualSlots.Count == 0)));
```

Comment:
Can the virtual slots be for some other method than Invoke?

Reply:
Thanks @jkotas, I am working on it.